### PR TITLE
Feat/add filters tests

### PR DESCRIPTION
**Pull request by WilsonNet**
# Add filter to tests and boots in tree details

- Changed filter lists to sets for better performance and clarity.
- Updated variable names for better readability.
- Added additional filters for configs, compiler, and architecture.
- Refactored the filtering logic in the `TreeDetailsSlow` view.
- Updated the frontend to handle new filter parameters.
- Removed unnecessary useSearch hook in BootsTab and TestsTab.

## How to test
Go to any tree that has tests eg: (broonie-sounds), and mess with the filters, see if it affects the boots/tests tab 

It is really hard to know if it is right, so, check the test details to see which filters you can use, because not all builds have tests

Coses #302, #326

### CODE REVIEW

File: backend/kernelCI_app/utils.py [hunk 1]
```diff
@@ -55,50 +55,54 @@ def getErrorResponseBody(reason: str):
...
+        "lte": ["lte", "<="],
     }
 
     def __init__(self, request):
         self.filters = []
         self.create_filters_from_req(request)
 
     def create_filters_from_req(self, request):
+        print(request.GET.keys())
+        print(request.GET)
```
Comment (lfjnascimento) [inline backend/kernelCI_app/utils.py:93]: leftover

File: backend/kernelCI_app/views/treeDetailsSlowView.py [hunk 2]
```diff
@@ -1,49 +1,82 @@
...
+                    filterBootStatus.add(value)
                 elif field == "boot.duration":
                     if operation == "lte":
                         filterBootDurationMax = toIntOrDefault(value, None)
                     else:
                         filterBootDurationMin = toIntOrDefault(value, None)
                 if field == "test.status":
-                    filterTestStatus = value
+                    filterTestStatus.add(value)
+                if field == "config_name":
```
Comment (lfjnascimento) [inline backend/kernelCI_app/views/treeDetailsSlowView.py:39]: Have you considered using a `switch/case`? I'm not sure, but maybe a `switch/case` would also help us not need the `# noqa: max-complexity=13`

File: backend/kernelCI_app/utils.py [hunk 3]
```diff
@@ -41,7 +41,7 @@ def extract_error_message(misc: Union[str, dict, None]):
     error_message = parsedEnv.get("error_msg")
     if error_message:
         return error_message
-    print("unknown error_msg in misc", misc)
+    # print("unknown error_msg in misc", misc)
```
Comment (loadez) [inline backend/kernelCI_app/utils.py:44]: remove unnecessary comment

File: backend/kernelCI_app/utils.py [hunk 4]
```diff
@@ -25,7 +25,7 @@ def extract_platform(misc_environment: Union[str, dict, None]):
     platform = parsedEnvMisc.get("platform")
     if platform:
         return platform
-    print("unknown platform in misc_environment", misc_environment)
+    # print("unknown platform in misc_environment", misc_environment)
```
Comment (loadez) [inline backend/kernelCI_app/utils.py:28]: remove unnecessary comment

File: backend/kernelCI_app/views/treeDetailsSlowView.py (hunk 2 above)
Comment (loadez) [reply backend/kernelCI_app/views/treeDetailsSlowView.py:39]: Heads up: switch/case is match in python.

The way to reduce complexity is to divide this function in smaller ones

File: backend/kernelCI_app/views/treeDetailsSlowView.py [hunk 5]
```diff
@@ -1,49 +1,82 @@
...
+            filterBootStatus,
+            filterBootDurationMin,
+            filterBootDurationMax,
+            filterTreeDetailsConfigs,
+            filterTreeDetailsCompiler,
+            filterArchitecture,
+        )
 
-    def get(self, request, commit_hash: str | None):
+    def get(self, request, commit_hash: str | None):  # noqa: max-complexity=13
```
Comment (loadez) [inline backend/kernelCI_app/views/treeDetailsSlowView.py:64]: ditto

File: backend/kernelCI_app/views/treeDetailsSlowView.py (hunk 2 above)
Comment (WilsonNet) [reply backend/kernelCI_app/views/treeDetailsSlowView.py]: it wasn't needed just splitted into functions

**Review by lucasbracher — APPROVED** [review]

**Review by loadez — APPROVED** [review]
LGTM

**Pull request merged**